PyDSS/ResultData.py

```diff
@@ -380,20 +368,24 @@ def deserialize(cls, data):
         return obj
 
     def add_values(self):
-        for prop in self.properties:
+        for  prop in self.properties:
             value = self._obj.GetValue(prop, convert=True)
-            # print(value)
-            # if isinstance(value, str):
-            #     quit()
-            # TODO: please check to this if this is correct
-            if self._data[prop] is None and not isinstance(value, str):
-                self._data[prop] = value
-            elif self._data[prop] is None and isinstance(value, str):
-                self._data[prop] = [value]
+            if isinstance(value, list):
+                for v in value:
+                    if self._data[prop] is None:
+                        self._data[prop] = v
+                    else:
+                        self._data[prop].append(v)
+                    if self._value_class is None:
+                        self.set_value_class(type(v))
```

Comment:
Under what scenario does this code block run?

Reply:
This will not run now given I have implemented ValueByList class.
Will remove it.

Reply:
done